cmd/controller-manager/app/controllermanager.go

```diff
@@ -0,0 +1,164 @@
+package app
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	restclient "k8s.io/client-go/rest"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/cli/globalflag"
+	"k8s.io/component-base/logs"
+	logsapi "k8s.io/component-base/logs/api/v1"
+	"k8s.io/component-base/term"
+	"k8s.io/component-base/version"
+	"k8s.io/component-base/version/verflag"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrlruntimecfg "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+
+	"github.com/karmada-io/karmada-operator/cmd/controller-manager/app/options"
+	operatorapi "github.com/karmada-io/karmada-operator/pkg/apis/operator/v1alpha1"
+	ctrlctx "github.com/karmada-io/karmada-operator/pkg/controller/context"
+	"github.com/karmada-io/karmada-operator/pkg/scheme"
+)
+
+func init() {
+	utilruntime.Must(logsapi.AddFeatureGates(utilfeature.DefaultMutableFeatureGate))
+}
+
+// NewControllerManagerCommand creates a *cobra.Command object with default parameters
+func NewControllerManagerCommand(ctx context.Context) *cobra.Command {
+	o := options.NewControllerManagerOptions()
+	cmd := &cobra.Command{
+		Use: "controller-manager",
+		PersistentPreRunE: func(*cobra.Command, []string) error {
+			// silence client-go warnings.
+			// karmada-operator generically watches APIs (including deprecated ones),
+			// and CI ensures it works properly against matching kube-apiserver versions.
+			restclient.SetDefaultWarningHandler(restclient.NoWarnings{})
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Fixme: define the karmada-operator's own verflag
+			verflag.PrintAndExitIfRequested()
+
+			// Activate logging as soon as possible, after that
+			// show flags with the final logging configuration.
+			if err := logsapi.ValidateAndApply(o.Logs, utilfeature.DefaultFeatureGate); err != nil {
+				return err
+			}
+
+			// To help debugging, print flags before further processing.
+			cliflag.PrintFlags(cmd.Flags())
```

Comment:
It is better to make it easy as far as possible. We can introduce them when we need them.